File: BlackJack/BlackJack_csharp/BlackJack/BlackJack/model/Deck.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BlackJack.model
{
	class Deck
	{
		List<Card> m_cards;
		Random r = new Random();

		public Deck()
		{
			m_cards = new List<Card>();

			for (int color = 0; color < (int)Card.Color.Count; color++)
			{
				for (int value = 0; value < (int)Card.Value.Count; value++)
				{
					Card c = new Card((Card.Color)color, (Card.Value)value);
					Add(c);
				}
			}
		}

		public Card GetCard()
		{
			Card c = m_cards.ElementAt(0);
			m_cards.RemoveAt(0);
			return c;
		}

		public void Add(Card a_card) {
			m_cards.Add(a_card);
		}

		public void Shuffle()
		{
			Random r = new Random();
			for (int i = 0; i < 1000; i++)
			{
				int ix = r.Next(m_cards.Count);
				Card c = m_cards.ElementAt(ix);
				m_cards.RemoveAt(ix);
				Add(c);
			}
		}
	}
}

File: BlackJack/BlackJack/model/Deck.java
package model;

import java.util.LinkedList;

public class Deck {

  LinkedList<Card> m_cards;

  public Deck() {
    m_cards = new LinkedList<Card>();
    
    for (int cIx = 0; cIx < Card.Color.Count.ordinal(); cIx++) {
      for (int vIx = 0; vIx < Card.Value.Count.ordinal(); vIx++) {
        Card c = new Card(Card.Color.values()[cIx], Card.Value.values()[vIx]);
        add(c);
      }
    }
  }
  
  private Card getRandomCard() {
    int index = (int)(Math.random() * 171717.0) % m_cards.size();
    Card c = m_cards.get(index);
    m_cards.remove(index);
    return c;
  }
  
  public void shuffle() {
    for (int i = 0; i < 1017; i++) {
          Card c = getRandomCard();
          add(c);
      }
  }
  
  private void add(Card c) {
    m_cards.add(c);
  }
}